finesse/hardware/plugins/temperature/senecak107.py

```diff
@@ -117,8 +117,7 @@ def parse_data(self, data: bytes) -> numpy.ndarray:
         # Converts incoming bytes into 16-bit ints
         ints = numpy.frombuffer(data, dt, 8, 3)
 
-        vals = self.calc_temp(ints)
-        return vals
+        return [self.calc_temp(val) for val in ints]
```

Comment:
This isn't the best way to make a new data structure here. It'd be better to keep the data in a numpy array. Also, the type hints are now wrong for `calc_temp()` although it will still work -- not sure why `mypy` isn't flagging it.

Instead, I'd change `calc_temp()` to assign its result to a new variable, rather than overwriting `vals`.

Reply:
I just noticed that the tests are failing and this is why.